index.js

```diff
@@ -0,0 +1,81 @@
+import service from './service';
+import config from './config';
+
+const instance = {};
+
+const init = async (configuration) => {
+	const {identifier, deregister} = await service(configuration);
+
+	Object.assign(instance, {identifier, deregister});
+
+	const {
+		properties,
+		register : registerConfigCallback,
+		deregister : deregisterConfigCallback,
+		stop : stopConfigPolling
+	} = await config(Object.assign({}, configuration, {service : identifier}));
+
+	Object.assign(instance, {
+		properties,
+		registerConfigCallback,
+		deregisterConfigCallback,
+		stopConfigPolling
+	});
+};
+
+/**
+ * Globally shared instance for accessing config and service. Will not be initialized until a call to `init` is made.
+ */
+export default instance;
```

Comment:
isnt it easier to return this from the init and mutate it from there? Now there is a global lock on it